### PR TITLE
Init with CI

### PR DESCRIPTION
**Pull request by jackjohn7**
Init Tauri, NextJS, and CI

### CODE REVIEW

File: client/eslint.config.mjs [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+
+const __filename = fileURLToPath(import.meta.url);
+const __dirname = dirname(__filename);
+
+const compat = new FlatCompat({
+  baseDirectory: __dirname,
+});
+
+const eslintConfig = [
+  ...compat.extends("next/core-web-vitals", "next/typescript"),
```
Comment (funnyboy-roks) [inline client/eslint.config.mjs:13]: Add the rules that we used in Channel Surfers:

```js
        rules: {
            camelcase: [
                'error',
                {
                    ignoreImports: true,
                },
            ],
            eqeqeq: 'error',
            'no-useless-assignment': 'warn',
            'consistent-return': 'error',
            'dot-notation': 'error',
            'no-unneeded-ternary': 'warn',
            'object-shorthand': 'error',
            'prefer-const': 'warn',
            'prefer-destructuring': 'error',
            'no-useless-rename': 'error',
            'no-cond-assign': 'error',
            '@typescript-eslint/no-unused-vars': [
                'error', // Change to 'error' if you want it to be an error
                {
                    args: 'all',
                    argsIgnorePattern: '^_',
                    caughtErrors: 'all',
                    caughtErrorsIgnorePattern: '^_',
                    destructuredArrayIgnorePattern: '^_',
                    varsIgnorePattern: '^_',
                    ignoreRestSiblings: true,
                },
            ],
        },
```

File: client/public/file.svg
Comment (funnyboy-roks) [inline client/public/file.svg:1]: We can remove all of the assets in the `client/public` folder.

**Review by funnyboy-roks — CHANGES_REQUESTED** [review]

**Comment by funnyboy-roks** [comment]
Can you also add prettier and the config that we used for Channel Surfers?

**Comment by funnyboy-roks** [comment]
Also should add our CI for eslint/prettier to this

File: .github/DOCS.md
[the remark this answers is not in the document]
Comment (jackjohn7) [reply .github/DOCS.md:1]: removed

File: client/public/file.svg
Comment (jackjohn7) [reply client/public/file.svg:1]: done

File: client/eslint.config.mjs (hunk 1 above)
Comment (jackjohn7) [reply client/eslint.config.mjs]: added

File: Cargo.toml
[the remark this answers is not in the document]
Comment (jackjohn7) [reply Cargo.toml:1]: added

**Review by funnyboy-roks — APPROVED** [review]

**Pull request merged**